scitokens-verify change breaks compatibility

#159 will cause existing scripts that use scitokens-verify (including [htdecodetoken](https://github.com/fermitools/htgettoken/blob/master/htdecodetoken)) to break.

Instead, the old form should be supported in addition to the new, at least for a transition period.  It could print a deprecation warning if the old form is used.

Ah, crash doesn't happen on el9, only el8.  On el8 there's also a compilation warning that may be relevant:
```
/home/dwd/work/scitokens-cpp/src/scitokens_internal.cpp: In function 'std::unordered_map<std::__cxx11::basic_string<char>, jwt::basic_claim<jwt::traits::kazuho_picojson> > json_to_claim_map(const object&)':
/home/dwd/work/scitokens-cpp/src/scitokens_internal.cpp:586:22: warning: structured bindings only available with -std=c++17 or -std=gnu++17
     for (const auto &[name, value] : json) {
                      ^
```

--- a/src/verify.cpp
+++ b/src/verify.cpp
@@ -90,10 +90,9 @@ int main(int argc, char *const *argv) {
     std::string token;
     // If a positional argument is present, treat it as the token (with warning)
     if (optind < argc) {
-        fprintf(stderr,
-                "%s: Warning: Providing the token on the command line is "
-                "insecure. Please use stdin instead.\n",
-                argv[0]);
+        std::cout << argv[0]
+                  << ": Warning: Providing the token on the command line is "
+                  << "insecure. Please use stdin instead." << std::endl;
         token = argv[optind];
     } else {
         // Read token from stdin

